File: PRACTICA1/RUBY/Deepspace/ShieldBooster.rb
#encoding:utf-8
require './ShieldToUI'
module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class ShieldBooster
  def initialize(name,boost,uses)
    @name = name #string
    @boost = boost #float
    @uses = uses #int
  end

  def name
    @name
  end

  def boost
    @boost
  end

  def uses
    @uses
  end

  def self.newCopy(orig)
    new(orig.name, orig.boost, orig.uses)
  end

  def getUIversion()
    ShieldToUI.new(self)
  end 

  def useIt

    if (@uses > 0)
      @uses = @uses -1
      return @boost
    else
      return 1.0
    end

  end

  def to_s
    return "Nombre: #{@name}, potenciador: #{@boost} y usos: #{@uses}\n "

  end 

end

end

File: PRACTICA1/JAVA/deepspace/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
    * @author Lucía Salamanca López
    *         Celia Arias Martínez
 */
class ShieldBooster implements CombatElement{
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String nom, float b, int us){
        name = nom;
        boost = b;
        uses = us;
    }
    
    ShieldBooster(ShieldBooster otro){
        name = otro.name;
        boost = otro.boost;
        uses = otro.uses;
    }
    
    public float getBoost(){
        return boost;
    }
    
    @Override
    public int getUses(){
        return uses;
    }
    
    @Override
    public float useIt(){
        if (uses > 0){
            uses--;
            return boost;
        }
        return 1.0f;            
    }
    
    public String toString(){
        String mensaje = "El ShieldBooster se llama "+name+", tiene de potenciador "+boost+" y tiene "+uses+" usos.";
        return mensaje;
    }
    
    ShieldToUI getUIversion(){
       return new ShieldToUI(this); 
    }
}
